code/drasil-code/lib/Language/Drasil/CodeSpec.hs

```diff
@@ -49,6 +49,12 @@ data CodeSpec where
   purpose :: Purpose,
   -- | Example Background.
   background :: Background,
+  -- | Example Scope.
```

Comment:
why are these all `Example X`? The data structure is a "Code Specification". It feels like there is a mismatch.

Futhermore, are we now requiring all this information for all codes?

Reply:
My guess is that these were meant to specify that it is the purpose, etc., of the example itself, but I think this is a bit redundant

Reply:
Maybe `CodeSpec` just needs to carry a reference to the `SystemInformation` it was constructed for? Right now, these fields are carried in both, and `CodeSpec`s are always a clone of `SystemInformation`s manually created ones.